### PR TITLE
C++ sol fix and py sol for cf-51c

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by freakin23** [comment]
AC (PY): https://codeforces.com/contest/51/submission/280631446
AC (C++): https://codeforces.com/contest/51/submission/280632444

File: solutions/silver/cf-51C.mdx [hunk 1]
```diff
@@ -88,6 +92,53 @@ int main() {
...
+EPSILON = 1e-7
+
+
+def test(d: int, x: list) -> int:
+	used = 0
+	pos = [x[0] + d] * 3
+	"""
+    Checks if each house is within range of the most recently placed
+    station, if not, builds a new one (unless there are none left)
+    """
```
Comment (SansPapyrus683) [inline solutions/silver/cf-51C.mdx:108]: weird formatting

Comment (freakin23) [reply solutions/silver/cf-51C.mdx:108]: pre-commit bot error.

Comment (freakin23) [reply solutions/silver/cf-51C.mdx:109]: I tried to add extra spacing to it but still don't know why it is showing like this here

Comment (TheGamingMousse) [reply solutions/silver/cf-51C.mdx:109]: given that it's only two lines, you could probably just use two individual comments (assuming precommit is still tweaking out)

Comment (SansPapyrus683) [reply solutions/silver/cf-51C.mdx]: no python doc comments are just like that
you need a string

Comment (SansPapyrus683) [reply solutions/silver/cf-51C.mdx]: > I tried to add extra spacing to it but still don't know why it is showing like this here

check ur tabs & spaces

Comment (freakin23) [reply solutions/silver/cf-51C.mdx]: corrected it directly from GitHub only.

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**